### ISSUE TITLE
Out of memory in android, after many calls to InitializeEverything

### ISSUE THREAD

**Issue by unitycoder** [unitycoder/UnityMobilePaint#19]
http://forum.unity3d.com/threads/released-mobile-paint-with-multi-touch.262645/page-4#post-2513834

**Comment by unitycoder** [comment]
texture amount gets increased each initialize call..
coming from this line:
`drawingTexture = new Texture2D(texWidth, texHeight, TextureFormat.RGBA32, false);`

**Comment by unitycoder** [comment]
temporary fix:
// FIND LINE
`drawingTexture = new Texture2D(texWidth, texHeight, TextureFormat.RGBA32, false);`

// CHANGE TO

```
if (drawingTexture==null)
{
    drawingTexture = new Texture2D(texWidth, texHeight, TextureFormat.RGBA32, false);
}
```

**Comment by unitycoder** [comment]
better fix, because the previous one wouldnt re-initialize it to new size etc.

// FIND LINE
`drawingTexture = new Texture2D(texWidth, texHeight, TextureFormat.RGBA32, false);`

// CHANGE TO

```
if (drawingTexture != null) Texture2D.DestroyImmediate(drawingTexture , true);
drawingTexture = new Texture2D(texWidth, texHeight, TextureFormat.RGBA32, false);
```

thanks to theLaymaster:
http://forum.unity3d.com/threads/released-mobile-paint-with-multi-touch.262645/page-4#post-2518199

**Comment by unitycoder** [comment]
fixed in the coming v1.96 update
